File: greetings-stat-service/stat-consumer/src/main/java/net/bakaar/greetings/stat/message/GreetingMessage.java
package net.bakaar.greetings.stat.message;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

import java.net.URI;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public record GreetingMessage(URI type, String payload) {
}

File: greetings-ui/src/models/GreetingMessage.ts
export class GreetingMessage {
    constructor(private readonly _id: string, private readonly _message: string, private readonly _type: string) {
    }

    public get message() {
        return this._message;
    }

    public get id() {
        return this._id;
    }

    public get type() {
        return this._type;
    }
}